substrate/frame/staking-async/src/tests/unbonding_queue.rs

```diff
@@ -0,0 +1,603 @@
+		.add_staker(102, 500, StakerStatus::<AccountId>::Nominator(vec![21]))
+		.add_staker(103, 500, StakerStatus::<AccountId>::Nominator(vec![31]))
+		.add_staker(104, 500, StakerStatus::<AccountId>::Nominator(vec![41]))
+		.add_staker(105, 500, StakerStatus::<AccountId>::Nominator(vec![51]))
+		.add_staker(106, 500, StakerStatus::<AccountId>::Nominator(vec![61]))
+		.add_staker(107, 500, StakerStatus::<AccountId>::Nominator(vec![71]))
+		.add_staker(108, 500, StakerStatus::<AccountId>::Nominator(vec![81]))
+		.add_staker(109, 500, StakerStatus::<AccountId>::Nominator(vec![91]))
+		.build_and_execute(|| {
+			assert_ok!(Staking::set_staking_configs(
+				RuntimeOrigin::root(),
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Set(UnbondingQueueConfig {
+					min_slashable_share: Perbill::from_percent(50),
+					lowest_ratio: Perbill::from_percent(34),
+					unbond_period_lower_bound: 1,
+					back_of_unbonding_queue: Zero::zero(),
+				})
+			));
+
+			assert_eq!(Staking::get_min_lowest_stake(), 0);
+			Session::roll_until_active_era(2);
+
+			// There are 9 validators, so one third of them would be 3.
+			// Hence, the lowest third would be composed of the following validators:
+			// 61 -> 100 (own) + 500 (nomination from 106)
+			// 71 -> 200 (own) + 500 (nomination from 107)
+			// 31 -> 500 (own) + 500 (nomination from 103)
+			// Summing up 2300 total.
+			assert_eq!(Staking::get_min_lowest_stake(), 2300);
+		});
+}
+
+#[test]
+fn calculate_lowest_total_stake_works() {
+	ExtBuilder::default().has_stakers(false).build_and_execute(|| {
+		assert_ok!(Staking::set_staking_configs(
+			RuntimeOrigin::root(),
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Set(UnbondingQueueConfig {
+				min_slashable_share: Perbill::from_percent(50),
+				lowest_ratio: Perbill::from_percent(34),
+				unbond_period_lower_bound: 1,
+				back_of_unbonding_queue: Zero::zero(),
+			})
+		));
+
+		Session::roll_until_active_era(2);
+		Session::roll_until_active_era(3);
+		Session::roll_until_active_era(4);
+		assert_eq!(current_era(), 4);
+		// There are no stakers
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![0, 0, 0]);
+		assert_eq!(Staking::get_min_lowest_stake(), 0);
+
+		// Create validators with different stakes for the next era.
+		let exposures = to_bounded_supports(vec![
+			(1, Support { total: 1000, voters: vec![] }),
+			(2, Support { total: 2000, voters: vec![] }),
+			(3, Support { total: 3000, voters: vec![] }),
+			(4, Support { total: 4000, voters: vec![] }),
+		]);
+
+		// Trigger new era to calculate the lowest proportion.
+		assert_ok!(EraElectionPlanner::<T>::do_elect_paged_inner(
+			exposures.clone().try_into().unwrap()
+		));
+		Session::roll_until_active_era(5);
```

Comment:
Yeah this is strange, the above line should not be needed.

Reply:
Why? The test fails if it gets removed.

Reply:
Test refactored in [4d21ca1](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/4d21ca1e36ea10cff917a9b7d97c3e02c82b636c).